src/tool/hpcrun/control-knob.h

```diff
@@ -4,6 +4,7 @@
 #define FORALL_KNOBS(macro)  \
   macro(HPCRUN_CUDA_DEVICE_BUFFER_SIZE)     \
   macro(HPCRUN_CUDA_DEVICE_SEMAPHORE_SIZE)  \
+	macro(STREAMS_PER_THREAD)	\
```

Comment:
Indentation

Reply:
> the control knobs still need work. I think that each sample source should register its control knobs with a control knob interface. That way, the control knobs available will be a function of the sample sources compiled in and available.

Ok, should it be done similarly like registering device-finalizers? What parameters, connected to Sample Sources, should be there? Should it be something like sampling frequency?

Reply:
things like sampling frequency should not be in control knobs. the nvidia settings used by hpctoolkit (e.g. device buffer size, semaphore count) are examples of control knobs. for the gpu tracing, we could have a control knob (for instance) for number of helper threads.

Reply:
Ok, I will try to implement that before our next meeting.

Reply:
it should be done similar to the device finalizers. each sample source can call register to register each knob. for nvidia sample source to register two knobs, it can just call register twice to register each of two knobs. the registry will hold a linked list of (name, value) pairs. the value can be represented by a void * in the registry. 

struct control_knob {
  char *name;
  void *(*value_parser)(char *value_string);
   char *(*value_printer)(void *value);
  void *value;
}

that way, a control knob can be anything: a string, an int, a double, or whatever. the void* can point to whatever storage is needed to hold the type.

Reply:
Thanks, will do it!